service/pricing/cid_gravity.go

```diff
@@ -84,24 +90,31 @@ func newClientRulesFor(apiKey, clientAddress string) *clientRules {
 // auction. The rules are cached locally for some time. It returns valid = false if the cached rules were expired but
 // couldn't be reloaded from the CID gravity API in time.
 func (cg *clientRules) PricesFor(auction *pb.Auction) (prices ResolvedPrices, valid bool) {
-	valid = cg.maybeReloadRules(cidGravityAPIUrl, cidGravityLoadRulesTimeout, cidGravityCachePeriod)
+	valid = cg.maybeReloadRules(cg.apiURL, cidGravityLoadRulesTimeout, cidGravityCachePeriod)
 	if !valid {
 		return
 	}
-	rules := cg.rules.Load()
+	var rules *rawRules
+
 	// preventive but should not happen
-	if rules == nil {
+	r := cg.rules.Load()
+	if r == nil {
 		valid = false
 		return
 	}
-	if rules.(*rawRules).Blocked {
+	rules = r.(*rawRules)
+
+	if rules.DealRateLimit > 0 && rules.CurrentDealRate >= rules.DealRateLimit {
+		return
```

Comment:
@merlinran, one question left here: how does `rules.DealRateLimit ` work here? I don't see this variable set anywhere but only in tests; maybe missing something? Shouldn't be updated when bidbot wins an auction?

Reply:
Or maybe is automatically fed by Lotus into cid-gravity.. so mostly outside of bidbot scope.
In theory whenever it wins a deal, we should propose a deal.. so that is fed by some Lotus cid-gravity plugin, and it would be updated that way?

Reply:
`DealRateLimit` is configured by the provider on the cid gravity console, and yeah, `CurrentDealRate` is updated by cid gravity server when it accepts a new deal by the lotus cid-gravity plugin.

<img width="355" alt="image" src="https://user-images.githubusercontent.com/1369696/139861916-97ba6411-311c-4aaa-9c33-4b5033696945.png">

It may be helpful to register one account on https://app.cidgravity.com to have a sense what it looks like.